apps/docs/content/guides/auth/third-party/firebase-auth.mdx

```diff
@@ -0,0 +1,254 @@
+---
+id: 'auth-third-party-firebase-auth'
+title: 'Firebase Auth'
+subtitle: 'Use Firebase Auth with your Supabase project'
+---
+
+<Admonition type="note">
+
+Creating a Firebase Auth third-party integration with your Supabase project is currently limited to a private alpha release. We are still trying to improve the developer experience around securing your project. Please reach out to [Supabase Support](https://supabase.help) if you're interested in having this enabled on your project.
+
+</Admonition>
+
+Firebase Auth can be used as a third-party authentication provider alongside Supabase Auth, or standalone, with your Supabase project.
+
+## Getting started
+
+1. First you need to add an integration to connect your Supabase project with your Firebase project. You will need to get the Project ID in the [Firebase Console](https://console.firebase.google.com/u/0/project/_/settings/general).
+2. Add a new Third-party Auth integration in your project's [Authentication settings](/dashboard/project/_/settings/auth).
+3. Create and attach restrictive RLS policies to all tables in your public schema, Storage and Realtime to prevent unauthorized access from unrelated Firebase projects.
+4. Assign the `role: 'authenticated'` [custom user claim](https://firebase.google.com/docs/auth/admin/custom-claims) to all your users.
+5. Finally set up the Supabase client in your application.
+
+<Tabs type="underlined" queryGroup="firebase-create-client">
+
+<TabPanel id="create-client-web" label="TypeScript">
+
+Creating a client for the Web is as easy as passing the `accessToken` async function. This function should [return the Firebase Auth JWT of the current user](https://firebase.google.com/docs/auth/admin/verify-id-tokens#web) (or null if no such user) is found.
+
+```typescript
+import { createClient } from '@supabase/supabase-js'
+
+const supabase = createClient('https://<supabase-project>.supabase.co', 'SUPABASE_ANON_KEY', {
+  accessToken: async () => {
+    ;(await firebase.auth().currentUser?.getIdToken(/* forceRefresh */ false)) ?? null
+  },
+})
+```
+
+Make sure the all users in your application have the `role: 'authenticated'` [custom claim](https://firebase.google.com/docs/auth/admin/custom-claims) set. If you're using the `onCreate` Cloud Function to add this custom role to newly signed up users, you will need to call `getIdToken(/* forceRefresh */ true)` immediately after sign up as the `onCreate` function does not run synchronously.
```

Comment:
since we already go over this below, i think we are repeating the same information here (oncreate not being synchronous etc). We could create a new heading called "Setting up the supabase client" and move this example there. 

That way getting started just contains the 5 bullet points, and it links to the 3 sections below
- Assign the "role" custom claim
- Setting up the supabase client
- Securing your project with restrictive RLS policies - This section could be moved all the way down since it is relevant for self hosting users only.